pkg/apis/core/v1alpha1/types_seed.go

```diff
@@ -124,18 +127,26 @@ type SeedDNS struct {
 	// IngressDomain is the domain of the Seed cluster pointing to the ingress controller endpoint. It will be used
 	// to construct ingress URLs for system applications running in Shoot clusters.
 	IngressDomain string `json:"ingressDomain" protobuf:"bytes,1,opt,name=ingressDomain"`
-	// NginxIngress controls settings for a gardener managed nginx-ingress controller
+	// Provider configures a DNSProvider
 	// +optional
-	NginxIngress *SeedSettingNginxIngress `json:"nginxIngress,omitempty" protobuf:"bytes,6,opt,name=nginxIngress"`
+	Provider *SeedDNSProvider `json:"provider,omitempty" protobuf:"bytes,2,opt,name=provider`
 }
 
-type SeedSettingNginxIngress struct {
-	// Enabled controls whether a gardener managed nginx-ingress is deployed. It is used to route external traffic to the Seed.
-	Enabled bool `json:"enabled" protobuf:"bytes,1,opt,name=enabled"`
-	// Provider describes the type of the dns-provider, for example `aws-route53`, see https://github.com/gardener/external-dns-management
-	Provider string `json:"provider" protobuf:"bytes,3,opt,name=provider"`
+// SeedDNSProvider configures a DNSProvider for Seeds
+type SeedDNSProvider struct {
+	// Type describes the type of the dns-provider, for example `aws-route53`
+	Type string `json:"type" protobuf:"bytes,1,opt,name=type"`
 	// SecretRef is a reference to a Secret object containing cloud provider credentials used for registering external domains.
-	SecretRef corev1.SecretReference `json:"secretRef" protobuf:"bytes,4,opt,name=secretRef"`
+	SecretRef corev1.SecretReference `json:"secretRef" protobuf:"bytes,2,opt,name=secretRef"`
+}
```

Comment:
> For the dnsProvider I did not follow the dnsProvider in the Shoot spec. I think it is a different use case in a Seed. We do neither have multiple domains nor do we have multiple providers and also do not plan to support that. 

There is no concrete use-case at this point in time, yes. However, this section here is mostly used to configure the `dnsv1alpha1.DNSProvider` object. Hence, let's please add the `domains` and `zones` fields like in the shoot DNS section, see also https://github.com/gardener/gardener/blob/master/charts/seed-dns/provider/templates/dnsprovider.yaml#L34-L53.
It doesn't "hurt" and it may be desired by people who have multiple DNS domains and hosted zones in one of their accounts but only want to activate one (or a subset) of them for Gardener.